### ISSUE TITLE
Возможность добавить список сайтов файлом

### ISSUE THREAD

**Issue by ZaMIk99** [Waujito/youtubeUnblock#106]
Есть возможность как-то добавить список сайтов в файле(сайт на каждой строке)
`procd_set_param command /usr/bin/youtubeUnblock --sni-domains=file`
или как-то так? Хочу подсунуть ему не ток ютуб, в --sni-domains добавил сайты и все работают, но мне надо много сайтов. Вставлять их через запятую когда есть готовые списки файлом сложно и неудобно.

**Comment by Waujito** [comment]
Да, можно добавить это. Пока что вы можете попробовать `cat fileee | sed -z 's/\n/,/g' | xargs sudo youtubeUnblock --sni-domains`
![image](https://github.com/user-attachments/assets/b8f8a994-f4fd-4251-a502-0bda0badd4c2)

**Comment by YBS-PC** [comment]
если я добавил меню для openwrt которое создает файл /opt/youtubeUnblock.txt с доменами которые я пропишу в окне создания файла а в файле /etc/init.d/youtubeUnblock (просто домены по одному в каждой строке)
```
googlevideo.com
ggpht.com
ytimg.com
youtube.com
play.google.com
youtu.be
googleapis.com
googleusercontent.com
gstatic.com
l.google.com
```
 пропишу такую строку procd_set_param command cat /opt/youtubeUnblock.txt | sed -z 's/\n/,/g' | xargs /usr/bin/youtubeUnblock --packet-mark=524288 --frag-sni-reverse=0 --silent --sni-domains=
все правильно будет работать?

**Comment by Waujito** [comment]
> если я добавил меню для openwrt которое создает файл /opt/youtubeUnblock.txt с доменами которые я пропишу в окне создания файла а в файле /etc/init.d/youtubeUnblock (просто домены по одному в каждой строке)
> 
> ```
> googlevideo.com
> ggpht.com
> ytimg.com
> youtube.com
> play.google.com
> youtu.be
> googleapis.com
> googleusercontent.com
> gstatic.com
> l.google.com
> ```
> 
> пропишу такую строку `procd_set_param command cat /opt/youtubeUnblock.txt | sed -z 's/\n/,/g' | xargs /usr/bin/youtubeUnblock --packet-mark=524288 --frag-sni-reverse=0 --silent --sni-domains=` все правильно будет работать?

Должно, но я не понимаю, зачем вы указываете все эти домены явно. Ровно такой же список идёт по дефолту --sni-domains вам не нужен.

**Comment by YBS-PC** [comment]
> Ровно такой же список идёт по дефолту

Это я для примера добавил. Так то там другой список будет. Только вопрос домены которые по умолчанию надо включать в этот список?

**Comment by Waujito** [comment]
Да, надо

**Comment by ZaMIk99** [comment]
Я пожалуй подожду более легкую реализацию, а не для хакеров с регулярками. Спешить некуда, всё работает.🌚
Спасибо за отзывчивость.

**Comment by Yusupoff** [comment]
Поставил пакеты работает отлично 
Вот с файлом доменов было бы интересно. нашел удобный список заблокированных доменов https://community.antifilter.download/list/domains.lst

**Comment by ZaMIk99** [comment]
в бетке в luci есть возможность добавлять домены, надеюсь когда будет возможность добавлять файлом, там тоже не будет всё на главной, а в отдельной кнопке, которая будет открывать файл для редактирования, а то сейчас после добавления десятка доменов выглядит ужасно

**Comment by vitidev** [comment]
Поддерживаю. Даже банальный textarea удобнее чем куча текстовых полей.

**Comment by jepleureturis** [comment]
Доброго времени! Сразу прошу прощения за глупые вопросы, с линуксом не особо дружен, но ваш способ обхода у меня заработал на ура (в отличии от того же zapret). Хотелось бы немного довести его до ума для своих нужд, но не очень понимаю каким образом. Имею Keenetic Extra, на нем установлен Entware (через флешку). В идеале - прописать все базовые домены (рутракер, инста, фейсбук и т.д) , чтобы система подтягивала их и пропускала без vpn. Насколько я понял в этой ветке - обход не предполагает список, который будет тянуть для искажения (как в том  же запрете). В связи с чем возникает вопрос - есть ли где-то в конфигах  список доменов, которые должны работать и если нет, то как в среде Entware реализовать нечто подобное? Попробовал команды, представленные выше, для OpenWRT - ожидаемо, что они не сработали. Буду благодарен, если поможете разобраться.

**Comment by Waujito** [comment]
> Доброго времени! Сразу прошу прощения за глупые вопросы, с линуксом не особо дружен, но ваш способ обхода у меня заработал на ура (в отличии от того же zapret). Хотелось бы немного довести его до ума для своих нужд, но не очень понимаю каким образом. Имею Keenetic Extra, на нем установлен Entware (через флешку). В идеале - прописать все базовые домены (рутракер, инста, фейсбук и т.д) , чтобы система подтягивала их и пропускала без vpn. Насколько я понял в этой ветке - обход не предполагает список, который будет тянуть для искажения (как в том  же запрете). В связи с чем возникает вопрос - есть ли где-то в конфигах  список доменов, которые должны работать и если нет, то как в среде Entware реализовать нечто подобное? Попробовал команды, представленные выше, для OpenWRT - ожидаемо, что они не сработали. Буду благодарен, если поможете разобраться. 
> 
> UPD: Открыл файл youtubeUnblock из папки bin через Notepad++ и через поиск нашел домены гугла и ютуба. Если я добавлю туда еще -   будет ли оно работать? UPD2: Попробовал - перестал и ютуб открываться, вернул все как было

Я не забыл про эту ишью, но всё никак не нахожу времени реализовать файлом, но после обновления системы аргументов в rc5 это сделать гораздо легче) 

Можно домены прописать флагом к анблоку —sni-domains. Посмотрите его в ридми. Попробуйте запустить анблок, только с этим флагом. Ютуб тоже там надо весь прописывать. Если всё работает - напишите флаг в /etc/init.d/youtubeUnblofk в args

**Comment by jepleureturis** [comment]
> запустить анблок, только с этим флагом

Если я верно вас понял, речь о команде youtubeunlock start и рядом прописывать флаг с доменом. Увы, затея не взлетела  :( В args тоже попробовал прописать флаг с единственным доменом rutracker.org - ничего не изменилось.

**Comment by jepleureturis** [comment]
@Waujito простите, что снова беспокою. Не сможете ли подсказать какие-то альтернативные решения или в чем моя ошибка и почему это не работает?)

**Comment by Waujito** [comment]
> > запустить анблок, только с этим флагом
> 
> Если я верно вас понял, речь о команде youtubeunlock start и рядом прописывать флаг с доменом. Увы, затея не взлетела :( В args тоже попробовал прописать флаг с единственным доменом rutracker.org - ничего не изменилось.

Необходимо просто запустить youtubeUnblock с этим флагом.
`/usr/bin/youtubeUnblock --sni-domains=...`

**Comment by jepleureturis** [comment]
> Необходимо просто запустить youtubeUnblock с этим флагом.
> `/usr/bin/youtubeUnblock --sni-domains=...`

Хм, по данному пути у меня есть только файл busybox.

Я запускаюсь из директории /opt/etc/init.d/S91youtubeUnblock. Но что-то явно делаю не так...![image](https://github.com/user-attachments/assets/d3d0557a-5694-417a-97d0-65a2b73801ba)

**Comment by Waujito** [comment]
> Я запускаюсь из директории /opt/etc/init.d/S91youtubeUnblock. Но что-то явно делаю не так...

/opt/use/bin/youtubeUnblock

**Comment by jepleureturis** [comment]
> /opt/usr/bin/youtubeUnblock

Я понимаю о чем вы, но такой файл в bin отсутствует. Т.е. он обязательно должен там быть? В моем варианте это работать не будет и нужно все переделать?)

**Comment by jepleureturis** [comment]
> After installation, the binary in /opt/bin and the init script in /opt/etc/init.d/S51youtubeUnblock will be available. To run the youtubeUnblock, simply run /opt/etc/init.d/S51youtubeUnblock start

Получается, что скрипт у меня появился,а бинарника нет

Что могло пойти не так в этом случае? Ведь бинарник сам функционирует

**Comment by Waujito** [comment]
> > After installation, the binary in /opt/bin and the init script in /opt/etc/init.d/S51youtubeUnblock will be available. To run the youtubeUnblock, simply run /opt/etc/init.d/S51youtubeUnblock start
> 
> Получается, что скрипт у меня появился,а бинарника нет
> 
> Что могло пойти не так в этом случае? Ведь скрипт сам функционирует

/opt/bin/youtubeUnblock?

**Comment by jepleureturis** [comment]
> /opt/bin/youtubeUnblock?

Увы, тоже нет

Папки usr у меня нет вообще, а в bin лежит только busybox...

**Comment by mj320981** [comment]
What do u want from me bro

🖤 1:<

________________________________
